adserver/tasks.py

```diff
@@ -114,71 +90,96 @@ def daily_update_geos(
             date__lt=end_date,  # Things at UTC midnight should count towards tomorrow
         ).delete()
 
-    for impression_type in IMPRESSION_TYPES:
-        topic_mapping = defaultdict(int)
-        queryset = _default_filters(impression_type, start_date, end_date)
-        for values in (
-            queryset.values("advertisement", "country", "publisher")
-            .annotate(total=Count("country"))
-            .filter(total__gt=0)
-            .order_by("-total")
-            .iterator()
-        ):
-            country = values["country"]
-            if geo:
-                impression, _ = GeoImpression.objects.using("default").get_or_create(
-                    publisher_id=values["publisher"],
-                    advertisement_id=values["advertisement"],
-                    country=country,
-                    date=start_date,
-                )
-                GeoImpression.objects.using("default").filter(pk=impression.pk).update(
-                    **{impression_type: values["total"]}
-                )
-
-            if region:
-
-                if country in us_ca:
-                    _region = "us-ca"
-                elif country in eu_aus_nz:
-                    _region = "eu-aus-nz"
-                elif country in wider_apac:
-                    _region = "wider-apac"
-                elif country in latin_america:
-                    _region = "latin-america"
-                elif country in africa:
-                    _region = "africa"
-                elif country in exclude:
-                    _region = "exclude"
-                elif country in south_asia:
-                    _region = "south-asia"
-                else:
-                    _region = "global"
-
-                publisher = values["publisher"]
-                advertisement = values["advertisement"]
-                topic_mapping[f"{advertisement}:{publisher}:{_region}"] += values[
-                    "total"
-                ]
+    topic_mapping = defaultdict(
+        lambda: {
+            "decisions": 0,
+            "offers": 0,
+            "views": 0,
+            "clicks": 0,
+        }
+    )
+    queryset = Offer.objects.using(settings.REPLICA_SLUG).filter(
```

Comment:
Might be good to keep all these queries the same, where we're just doing a single filter on `Offer` as done below: https://github.com/readthedocs/ethical-ad-server/pull/532/files#diff-bdef58a59b4162498b09457b29f37148cb2494ada036cc9ba8293f625b14b1bdR196-R214